design/aws-ecs-scheduled-ecs-task-construct.md

```diff
@@ -0,0 +1,131 @@
+# AWS ECS - L3 Construct for Scheduling EC2 Tasks
+
+To address issue [#2352](https://github.com/awslabs/aws-cdk/issues/2352), the ECS CDK construct library should provide a way for customers to create a standalone scheduled task, without creating a service, for their container. The task will be initiated by a cloudwatch event that is scheduled based on [CW Scheduled Events](http://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html).
+
+This would mean adding a new ECS CDK construct `ScheduledEc2Task`, that would take in the necessary properties required to create a Task Definition, an EventRuleTarget as well as an AWS EventRule.
+
+Note: Currently, ScheduledTasks, via CloudFormation, are only supported for EC2 and not for Fargate. CloudFormation does not support every [EcsParameter](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-events-rule-ecsparameters.html) that is supported by the [Amazon CloudWatch Events API](https://docs.aws.amazon.com/AmazonCloudWatchEvents/latest/APIReference/API_EcsParameters.html). It is currently missing the following custom parameters:
+
+* Group
+* LaunchType
+* NetworkConfiguration
+* PlatformVersion
+
+TaskDefinitions contain a NetworkMode, and EC2 can support none, bridge, host or awsvpc (which requires a NetworkConfiguration attribute). However, TaskDefinitions running on Fargate can only support a NetworkMode of awsvpc. For this construct, we're creating a scheduled task (that is standalone task, not backed with a service). It works for ECS because we're using bridge as the NetworkMode, which does not require a NetworkConfiguration to be present.
+
+## General approach
+
+The new [`ecs.ScheduledEc2Task`] class will include an L3 construct for:
+
+* ScheduledEc2Task
+
+A `ScheduledEc2Task` will create a task definition with the specified container. An `Ec2EventRuleTarget` will be created and associated as the target to an `Amazon Cloudwatch Event Rule` (indicating how frequently the task should be run). Based on the `Amazon Cloudwatch Event Rule` schedule, a task will run on the EC2 instances specified in the cluster. 
+
+## Code changes
+
+Given the above, we should make the following changes to support scheduled tasks on ECS:
+1. Create `ScheduledEc2TaskProps` interface  and `ScheduledEc2Task` construct
+
+# Part 1: Create `ScheduledEc2TaskProps` interface  and `ScheduledEc2Task` construct  
+
+The `ScheduledEc2TaskProps` interface will contain properties to construct the Ec2TaskDefinition, Ec2EventRuleTarget and EventRule:
+
+```ts
+export interface ScheduledEc2TaskProps {
+  /**
+   * The cluster where your service will be deployed.
+   */
+  readonly cluster: ICluster;
+
+  /**
+   * The CMD value to pass to the container.
+   *
+   * If you provide a shell command as a single string, you have to quote command-line arguments.
+   */
+  readonly command?: string[];
```

Comment:
This is getting nit-picky now, but I think we should be able to make this a nicer experience.  For example, the console allows entering this as a comma-delimited string instead of string array

Reply:
Do we want to mimic the behaviour that is used in the console? I can have the props accept a string and convert it to a string array when i add it to the taskdef?

Reply:
Yeah that's what I was thinking

Reply:
Makes sense to me, changes have been made.